src/vivarium_gates_mncnh/components/mortality.py

```diff
@@ -264,9 +264,9 @@ def on_time_step(self, event: Event) -> None:
 
         # Determine which neonates die and update metadata
         dead_idx = self.randomness.filter_for_probability(
-            alive_children.index,
+            alive_idx,
             mortality_risk,
-            f"{self._sim_step_name}_neonatal_mortality_choice",
+            f"{self._sim_step_name()}_choice",
```

Comment:
This is the bugfix. The rest of it is just name updates and additional checks to ensure the model is working correctly.

Reply:
Passing the callable instead of the output of the callable gets hashed differently